cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_impl_dpc.cpp

```diff
@@ -0,0 +1,1230 @@
+    Float mrgmean = Float(0);
+
+    if constexpr (check_mask_flag(bs_list::mean | sum2cent_based_stat, List)) {
+        lrc_ptr[id] = 0;
+    }
+
+    for (std::int64_t i = id; i < block_count; i += local_size) {
+        std::int64_t offset = group_id * block_count + i;
+
+        Float min = Float(0);
+        if constexpr (check_mask_flag(bs_list::min, List)) {
+            min = bmin_ptr[offset];
+        }
+        Float max = Float(0);
+        if constexpr (check_mask_flag(bs_list::max, List)) {
+            max = bmax_ptr[offset];
+        }
+        Float sum = Float(0);
+        if constexpr (check_mask_flag(bs_list::sum | bs_list::mean | sum2cent_based_stat, List)) {
+            sum = bsum_ptr[offset];
+        }
+        Float sum2 = Float(0);
+        if constexpr (check_mask_flag(bs_list::sum2 | bs_list::sorm, List)) {
+            sum2 = bsum2_ptr[offset];
+        }
+        std::int64_t rcnt = 1;
+        if constexpr (check_mask_flag(bs_list::mean | sum2cent_based_stat, List)) {
+            rcnt = brc_ptr[offset];
+        }
+        Float sum2cent = Float(0);
+        if constexpr (check_mask_flag(sum2cent_based_stat, List)) {
+            sum2cent = bsum2cent_ptr[offset];
+        }
+        Float mean = sum / static_cast<Float>(rcnt);
+
+        Float sum_n1n2 = mrgvectors + static_cast<Float>(rcnt);
+        Float mul_n1n2 = mrgvectors * static_cast<Float>(rcnt);
+        Float delta_scale = mul_n1n2 / sum_n1n2;
+        Float mean_scale = Float(1) / sum_n1n2;
+        Float delta = mean - mrgmean;
+
+        mrgmin = sycl::fmin(min, mrgmin);
+        mrgmax = sycl::fmax(max, mrgmax);
+        mrgsum += sum;
+        mrgsum2 += sum2;
+        mrgsum2cent = mrgsum2cent + sum2cent + delta * delta * delta_scale;
+        mrgmean = (mrgmean * mrgvectors + mean * static_cast<Float>(rcnt)) * mean_scale;
+        mrgvectors = sum_n1n2;
+
+        if constexpr (check_mask_flag(bs_list::min, List)) {
+            lmin_ptr[id] = mrgmin;
+        }
+        if constexpr (check_mask_flag(bs_list::max, List)) {
+            lmax_ptr[id] = mrgmax;
+        }
+        if constexpr (check_mask_flag(bs_list::sum | bs_list::mean | sum2cent_based_stat, List)) {
+            lsum_ptr[id] = mrgsum;
+        }
+        if constexpr (check_mask_flag(bs_list::sum2 | bs_list::sorm, List)) {
+            lsum2_ptr[id] = mrgsum2;
+        }
+        if constexpr (check_mask_flag(bs_list::mean | sum2cent_based_stat, List)) {
+            lrc_ptr[id] += rcnt;
+        }
+        if constexpr (check_mask_flag(sum2cent_based_stat, List)) {
+            lsum2cent_ptr[id] = mrgsum2cent;
+        }
+        if constexpr (check_mask_flag(bs_list::mean | sum2cent_based_stat, List)) {
+            lmean_ptr[id] = mrgmean;
+        }
+    }
+
+    for (std::int64_t stride = sycl::min(local_size, block_count) / 2; stride > 0; stride /= 2) {
+        item.barrier(sycl::access::fence_space::local_space);
+
+        if (stride > id) {
```

Comment:
Please don't do this. Please use a common trick instead:
```suggestion
        if (id >= stride) 
            return;
```

Reply:
The problem of this trick is that you can't just go out from scope for certain workitem in case if you have a collective operation in this scope, because other WI will wait for the one which go away.
With this trick kernel is hanging one.